datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -300,6 +306,130 @@ fn cast_static_filter_to_set(list: &[Arc<dyn PhysicalExpr>]) -> HashSet<ScalarVa
     }))
 }
 
+fn make_list_contains_decimal(
+    array: &DecimalArray,
+    list: Vec<ColumnarValue>,
+    negated: bool,
+) -> BooleanArray {
+    let contains_null = list
+        .iter()
+        .any(|v| matches!(v, ColumnarValue::Scalar(s) if s.is_null()));
+    let values = list
+        .iter()
+        .flat_map(|v| match v {
+            ColumnarValue::Scalar(s) => match s {
+                Decimal128(v128op, _, _) => v128op.map(|v128| v128),
```

Comment:
I don't understand what this `map` does. Is this the same?

```suggestion
                Decimal128(v128op, _, _) => *v128op,
```

Reply:
the type of v128op  is `&Option`, we just need the Some value.
Just like that
```
match v128op {
  None => None
 Some(value) => Some(value)
}
```